generate_markdown_from_doxygen_xml.py

```diff
@@ -0,0 +1,1008 @@
+"""
+This script converts Doxygen XML output into gitbook-compatible markdown. There are no arguments.
+
+- The directories that are parsed for source/outputs is DOXYGEN_XML_DIR (set DOXYGEN_ROOT_DIR for path from this file execution)
+- The list of files to parse are in the set: FILES_TO_PROCESS
+
+How it works:
+- There are Python objects for each C++ type - e.g. cppClass, cppTypdef etc. (cppClass covers both class and struct)
+- Each object has a method import_from_doxygen_class_file() which takes a Tag (or filename for CPP) and parses in data. 
+  - The cpp class takes a file, and creates the sub objects it owns (e.g. methods), calling their import_from_doxygen_class_file() method with a tag.
+- Each object (mostly) has a markdown_overview() and markdown() method that renders the markdown from the respective objects
+- markdown_any_tag() - 
+  - Some objects/values will be empty and others will store tagged information that is only partially parsed from XML (in theory this allows us 
+  to change the output rendering, because we store it in generic format. The markdown_any_tag() method can be used to get 
+  markdown text or an empty string from any stored attribute. 
+  - This can be safely called on any "data" type attributed
+  - This will report an error on console if an unsupported tag is detected. The tag will render in output anyway, but may contain odd tags.
+- cleanup_markdown_string
+  - Just cleans up a little of the markdown.
+
+"""
+
+import argparse
+import os
+import xml.etree.ElementTree as ET 
+
+
+# Load all the xml files in the dir 
+DOXYGEN_ROOT_DIR= './install/docs'
+DOXYGEN_XML_DIR = DOXYGEN_ROOT_DIR + '/xml'
+DOXYGEN_OUTPUT_DIR = DOXYGEN_ROOT_DIR + '/markdown'
+
+
+FILES_TO_PROCESS = {
+    'classdronecore_1_1_action.xml', 
+    'classdronecore_1_1_drone_core.xml',
+    'classdronecore_1_1_device.xml',
+    'classdronecore_1_1_info.xml',
+    'classdronecore_1_1_logging.xml',
+    'classdronecore_1_1_mission.xml',
+    'classdronecore_1_1_mission_item.xml',
+    'classdronecore_1_1_offboard.xml',
+    'classdronecore_1_1_telemetry.xml',
+    'structdronecore_1_1_telemetry_1_1_battery.xml',
+    'structdronecore_1_1_telemetry_1_1_euler_angle.xml',
+    'structdronecore_1_1_telemetry_1_1_g_p_s_info.xml',
+    'structdronecore_1_1_telemetry_1_1_ground_speed_n_e_d.xml',
+    'structdronecore_1_1_telemetry_1_1_health.xml',
+    'structdronecore_1_1_telemetry_1_1_position.xml',
+    'structdronecore_1_1_telemetry_1_1_quaternion.xml',
+    'structdronecore_1_1_telemetry_1_1_r_c_status.xml',
+    'structdronecore_1_1_info_1_1_version.xml',
+    'structdronecore_1_1_offboard_1_1_velocity_body_yawspeed.xml',
+    'structdronecore_1_1_offboard_1_1_velocity_n_e_d_yaw.xml',
+    'classdronecore_1_1_device_plugin_container.xml',
+    'classdronecore_1_1_plugin_base.xml'
```

Comment:
Would be nice to have this auto-generated otherwise it will break at some point.

Reply:
OK, now runs over all files in XML folder

Reply:
Ok, how come this file list is still here?

Reply:
Removed. It was left in, commented out.